p2p/connection.go

```diff
@@ -553,29 +553,27 @@ func (chDesc *ChannelDescriptor) FillDefaults() {
 type Channel struct {
 	conn          *MConnection
 	desc          *ChannelDescriptor
-	id            byte
 	sendQueue     chan []byte
 	sendQueueSize int32 // atomic.
 	recving       []byte
 	sending       []byte
-	priority      int
 	recentlySent  int64 // exponential moving average
 
 	maxMsgPacketPayloadSize int
 }
 
 func newChannel(conn *MConnection, desc *ChannelDescriptor) *Channel {
+	descCopy := *desc
```

Comment:
i want to protect the case where the caller happens to incorrectly use the same `ChannelDescriptor` for two different `newChannel` calls and mutates it inbetween. Then they could both be sharing the same ChannelDescriptor and I want to avoid that possibility even in this same source file.

Reply:
also, i felt that by copying i was actually avoiding modifying the parameter potentially later in `FillDefaults`

Reply:
```
descCopy := *desc
descCopy.FillDefaults()

return &Channel{
  		conn: conn,
                desc: descCopy,
```

?

Reply:
can't we just pass in a non-pointer directly in `newChannel`

Reply:
i do not like the copy forms when we can instead just pass a non-pointer directly. i think that is better because it lets the reader know the method is pure without reading the method body. what do you think of @ebuchman suggestion @melekes  ?

Reply:
if we can pass in a non-pointer, let's do that

Reply:
Let's just make ChannelDescriptor a non-pointer everywhere. It's a tiny struct anyways. Only thing is `.FillDefaults` will then have to return a copy, but thats fine.